src/octree/generation.rs

```diff
@@ -199,27 +214,33 @@ fn subsample_children_into(
             Err(Error(ErrorKind::NodeNotFound, _)) => continue,
             Err(err) => return Err(err),
         };
-        let point_iterator = PointIterator::from_data_provider(
+        let mut node_iterator = NodeIterator::from_data_provider(
             octree_data_provider,
             octree_meta.encoding_for_node(child_id),
             &child_id,
             num_points as usize,
+            NUM_POINTS_PER_BATCH,
         )?;
 
         // We read all points into memory, because the new node writer will rewrite this child's
         // file(s).
-        let mut points = Vec::with_capacity(point_iterator.size_hint().1.unwrap());
-        point_iterator.for_each(|p| points.push(p));
+        let mut batch = node_iterator.next().unwrap();
+        node_iterator.for_each(|mut b| batch.append(&mut b).unwrap());
+        let (keep_parent, keep_child): (Vec<bool>, Vec<bool>) = (0..batch.position.len())
+            .map(|i| {
+                let in_parent = i % 8 == 0;
+                (in_parent, !in_parent)
+            })
+            .unzip();
+        let mut parent_batch = batch.clone();
```

Comment:
Same considerations for efficiency here, but optional unless you think this is a critical section

Reply:
See comment above